bot/exts/evergreen/profile_pic_modification/pfp_modify.py

```diff
@@ -0,0 +1,272 @@
+            )
+
+            embed = discord.Embed(
+                name="Your Lovely Pride Avatar!",
+                description=f"Here is your lovely avatar, surrounded by\n a beautiful {option} flag. Enjoy :D"
+            )
+            embed.set_image(url=f"attachment://{file_name}")
+            embed.set_footer(text=f"Made by {ctx.author.display_name}.", icon_url=ctx.author.avatar_url)
+            await ctx.send(file=file, embed=embed)
+
+    @pfp_modify.group(
+        aliases=["avatarpride", "pridepfp", "prideprofile"],
+        root_aliases=("prideavatar", "avatarpride", "pridepfp", "prideprofile"),
+        invoke_without_command=True
+    )
+    async def prideavatar(self, ctx: commands.Context, option: str = "lgbt", pixels: int = 64) -> None:
+        """
+        This surrounds an avatar with a border of a specified LGBT flag.
+
+        This defaults to the LGBT rainbow flag if none is given.
+        The amount of pixels can be given which determines the thickness of the flag border.
+        This has a maximum of 512px and defaults to a 64px border.
+        The full image is 1024x1024.
+        """
+        option = option.lower()
+        pixels = max(0, min(512, pixels))
+        flag = GENDER_OPTIONS.get(option)
+        if flag is None:
+            await ctx.send("I don't have that flag!")
+            return
+
+        async with ctx.typing():
+            image_bytes = await ctx.author.avatar_url.read()
+            await self.send_pride_image(ctx, image_bytes, pixels, flag, option)
+
+    @prideavatar.command()
+    async def image(self, ctx: commands.Context, url: str, option: str = "lgbt", pixels: int = 64) -> None:
+        """
+        This surrounds the image specified by the URL with a border of a specified LGBT flag.
+
+        This defaults to the LGBT rainbow flag if none is given.
+        The amount of pixels can be given which determines the thickness of the flag border.
+        This has a maximum of 512px and defaults to a 64px border.
+        The full image is 1024x1024.
+        """
+        option = option.lower()
+        pixels = max(0, min(512, pixels))
+        flag = GENDER_OPTIONS.get(option)
+        if flag is None:
+            await ctx.send("I don't have that flag!")
+            return
+
+        async with ctx.typing():
+            async with self.bot.http_session as session:
+                try:
+                    response = await session.get(url)
+                except client_exceptions.ClientConnectorError:
+                    raise BadArgument("Cannot connect to provided URL!")
+                except client_exceptions.InvalidURL:
+                    raise BadArgument("Invalid URL!")
+
+                if response.status != 200:
+                    await ctx.send("Bad response from provided URL!")
+                    return
+
+                image_bytes = await response.read()
+                await self.send_pride_image(ctx, image_bytes, pixels, flag, option)
```

Comment:
Why are we closing the bot's `http_session` here?

Reply:
Good spot, the previous cog I pulled this from made a new session here, so I changed it to the bot's session, without thinking about it too much!